src/coalesce.c

```diff
@@ -0,0 +1,176 @@
+#include "data.table.h"
+
+void which_int(int *x, int nx, int *out, int *nout, int val) {
+  int j=0;
+  for (int i=0; i<nx; i++) {
+    if (x[i] == val) {
+      out[j] = i;
+      j++;
+    }
+  }
+  nout[0] = j+1;
+}
+void which_double(double *x, int nx, int *out, int *nout, double val) {
+  int j=0;
+  if (ISNA(val)) {
+    for (int i=0; i<nx; i++) {
+      if (ISNA(x[i])) {
+        out[j] = i;
+        j++;
+      }
+    }
+  } else {
+    for (int i=0; i<nx; i++) {
+      if (x[i] == val) {
```

Comment:
Quoting Matt:
> What if the integer64 value holds an integer which is equivalent to NaN in double?  I don't think it works then does it?,  returning NaN rather than false.

so we might need to have `which_int64` and cast `(int_64_t *) REAL(x)` for that rather than re-using this one.

Reply:
as of now made using casting pointers